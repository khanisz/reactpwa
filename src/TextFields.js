import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import Icon from "@material-ui/core/Icon";

const styles = theme => ({
  container: {
    display: "flex",
    flexWrap: "wrap"
  },
  textField: {
    marginLeft: theme.spacing.unit,
    marginRight: theme.spacing.unit,
    width: 200
  },
  dense: {
    marginTop: 19
  },
  menu: {
    width: 200
  },
  pos: {
    padding: 4
  }
});

const currencies = [
  {
    value: "USD",
    label: "$"
  },
  {
    value: "EUR",
    label: "€"
  },
  {
    value: "BTC",
    label: "฿"
  },
  {
    value: "JPY",
    label: "¥"
  }
];

class TextFields extends React.Component {
  state = {
    name: "",
    age: "",
    multiline: "Controlled",
    currency: "EUR",
    answered: false
  };

  handleChange = name => event => {
    this.setState({
      [name]: event.target.value
    });
  };

  render() {
    const { classes } = this.props;
    const { siema } = this.props;
    let texty;

    if (this.state.name=="siema"){
      this.state.answered = true;
    }
    if (this.state.answered) {
      texty = <div>Odpowiedź: {siema}</div>;
    } else {
      texty = <div/>;
    }

    return (
      <div>
        <div>
          <form className={classes.container} noValidate autoComplete="off">
            <TextField
              id="standard-name"
              label="Hasło"
              className={classes.textField}
              value={this.state.name}
              onChange={this.handleChange("name")}
              margin="normal"
            />
          </form>
        </div>
        <div>{texty}</div>
      </div>
    );
  }
}

TextFields.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(TextFields);
