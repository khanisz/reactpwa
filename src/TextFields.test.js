import React from "react";
import { shallow } from "enzyme";
import TextFields from "./TextFields";

describe("TextComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<TextFields debug />);

    expect(component).toMatchSnapshot();
  });

  it("should render correctly with no props", () => {
    const component = shallow(<TextFields />);

    expect(component).toMatchSnapshot();
  });

  it("should render correctly with title", () => {
    const answer = "Tytul";
    const component = shallow(<TextFields answer={answer} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with text", () => {
    const resolution = "Lore ipsum";
    const component = shallow(<TextFields resolution={resolution} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with title and text", () => {
    const answer = "Tytul";
    const resolution = "Lore ipsum";
    const component = shallow(
      <TextFields answer={answer} resolution={resolution} />
    );
    expect(component).toMatchSnapshot();
  });
});
