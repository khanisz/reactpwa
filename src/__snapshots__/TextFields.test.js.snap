// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextComponent should render correctly in "debug" mode 1`] = `
<TextFields
  classes={
    Object {
      "container": "TextFields-container-1",
      "dense": "TextFields-dense-3",
      "menu": "TextFields-menu-4",
      "pos": "TextFields-pos-5",
      "textField": "TextFields-textField-2",
    }
  }
  debug={true}
/>
`;

exports[`TextComponent should render correctly with no props 1`] = `
<TextFields
  classes={
    Object {
      "container": "TextFields-container-1",
      "dense": "TextFields-dense-3",
      "menu": "TextFields-menu-4",
      "pos": "TextFields-pos-5",
      "textField": "TextFields-textField-2",
    }
  }
/>
`;

exports[`TextComponent should render correctly with text 1`] = `
<TextFields
  classes={
    Object {
      "container": "TextFields-container-1",
      "dense": "TextFields-dense-3",
      "menu": "TextFields-menu-4",
      "pos": "TextFields-pos-5",
      "textField": "TextFields-textField-2",
    }
  }
  resolution="Lore ipsum"
/>
`;

exports[`TextComponent should render correctly with title 1`] = `
<TextFields
  answer="Tytul"
  classes={
    Object {
      "container": "TextFields-container-1",
      "dense": "TextFields-dense-3",
      "menu": "TextFields-menu-4",
      "pos": "TextFields-pos-5",
      "textField": "TextFields-textField-2",
    }
  }
/>
`;

exports[`TextComponent should render correctly with title and text 1`] = `
<TextFields
  answer="Tytul"
  classes={
    Object {
      "container": "TextFields-container-1",
      "dense": "TextFields-dense-3",
      "menu": "TextFields-menu-4",
      "pos": "TextFields-pos-5",
      "textField": "TextFields-textField-2",
    }
  }
  resolution="Lore ipsum"
/>
`;
